test/cpp/ext/otel/otel_tracing_test.cc

```diff
@@ -0,0 +1,545 @@
+      outbound_seq_nums.push_back(
+          std::get<uint64_t>(event.GetAttributes().at("sequence-number")));
+    }
+    if (event.GetName() == "Inbound message") {
+      inbound_seq_nums.push_back(
+          std::get<uint64_t>(event.GetAttributes().at("sequence-number")));
+    }
+  }
+  EXPECT_THAT(outbound_seq_nums,
+              UnorderedElementsAre(0, 1, 2, 3, 4, 5, 6, 7, 8, 9));
+  EXPECT_THAT(inbound_seq_nums,
+              UnorderedElementsAre(0, 1, 2, 3, 4, 5, 6, 7, 8, 9));
+  const auto server_span = std::find_if(
+      spans.begin(), spans.end(), [](const std::unique_ptr<SpanData>& span) {
+        return span->GetName() ==
+               "Recv.grpc.testing.EchoTestService/BidiStream";
+      });
+  outbound_seq_nums.clear();
+  inbound_seq_nums.clear();
+  // Verify messages on the server span
+  for (const auto& event : (*server_span)->GetEvents()) {
+    if (event.GetName() == "Outbound message") {
+      outbound_seq_nums.push_back(
+          std::get<uint64_t>(event.GetAttributes().at("sequence-number")));
+    }
+    if (event.GetName() == "Inbound message") {
+      inbound_seq_nums.push_back(
+          std::get<uint64_t>(event.GetAttributes().at("sequence-number")));
+    }
+  }
+  EXPECT_THAT(outbound_seq_nums,
+              UnorderedElementsAre(0, 1, 2, 3, 4, 5, 6, 7, 8, 9));
+  EXPECT_THAT(inbound_seq_nums,
+              UnorderedElementsAre(0, 1, 2, 3, 4, 5, 6, 7, 8, 9));
+  EXPECT_NE(server_span, spans.end());
+}
+
+TEST_F(OTelTracingTest, Retries) {
+  {
+    ChannelArguments args;
+    args.SetString(GRPC_ARG_SERVICE_CONFIG,
+                   "{\n"
+                   "  \"methodConfig\": [ {\n"
+                   "    \"name\": [\n"
+                   "      { \"service\": \"grpc.testing.EchoTestService\" }\n"
+                   "    ],\n"
+                   "    \"retryPolicy\": {\n"
+                   "      \"maxAttempts\": 3,\n"
+                   "      \"initialBackoff\": \"0.1s\",\n"
+                   "      \"maxBackoff\": \"120s\",\n"
+                   "      \"backoffMultiplier\": 1,\n"
+                   "      \"retryableStatusCodes\": [ \"ABORTED\" ]\n"
+                   "    }\n"
+                   "  } ]\n"
+                   "}");
+    auto channel = CreateCustomChannel(server_address_,
+                                       InsecureChannelCredentials(), args);
+    auto stub = EchoTestService::NewStub(channel);
+    EchoRequest request;
+    request.set_message("foo");
+    request.mutable_param()->mutable_expected_error()->set_code(
+        StatusCode::ABORTED);
+    EchoResponse response;
+    grpc::ClientContext context;
+    grpc::Status status = stub->Echo(&context, request, &response);
+  }
+  absl::SleepFor(absl::Milliseconds(500 * grpc_test_slowdown_factor()));
+  auto spans = data_->GetSpans();
+  EXPECT_EQ(spans.size(), 7);  // 1 client span, 3 attempt spans, 3 server spans
+  std::vector<uint64_t> attempt_seq_nums;
```

Comment:
Same as above: Suggest using `std::set<>` here, so that we can use `ElementsAre()` below.

Reply:
I don't see the reason to use an std::set. We can use `ElementsAre()` irrespective.